### PR TITLE
Add details of NIC requirements

### PR DESCRIPTION
**Pull request by w13915984028**
Issue: https://github.com/harvester/harvester/issues/5509

### CODE REVIEW

**Comment by SlavikCA** [comment]
Question about the config of 2 nodes + witness: 

Does the witness node both `management cluster network` and `VM workload network`? Or only `management cluster network`?

**Comment by starbops** [comment]
> Question about the config of 2 nodes + witness:
> 
> Does the witness node needs both management cluster network and VM workload network? Or only > management cluster network?

VMs will not be scheduled for the witness node, so only the `management cluster network` would be fine. Other nodes could have the VM workload network configured.

File: docs/install/requirements.md [hunk 1]
```diff
@@ -28,12 +28,14 @@ Harvester nodes have the following hardware requirements and recommendations for
...
 
 :::info important
 - Use server-class hardware to achieve the best results. Laptops and nested virtualization are not supported.
 - Each node must have a unique `product_uuid` (fetched from `/sys/class/dmi/id/product_uuid`) to prevent errors from occurring during VM live migration and other operations. For more information, see [Issue #4025](https://github.com/harvester/harvester/issues/4025).
+- Harvester has a [built-in management cluster network](../networking/clusternetwork.md#built-in-cluster-network). To achieve the best performance and high-availability. The general practice in production is to use at least 2 NICs in each node to setup a bonded NIC for the management network. Refer the step 6 of the [ISO Installation](../install/iso-install.md#installation-steps). You may also create additional [customized cluster network](../networking/clusternetwork.md#custom-cluster-network) to carry the VM workload network separately. Each of them requires at least 2 additional NICs to setup a bonded NIC in every involved node of the Harvester cluster. You may refer [Cluster Network](../networking/clusternetwork.md#concepts) for more details.
```
Comment (starbops) [inline docs/install/requirements.md:37]: > To achieve the best performance and high-availability.

This sentence seems to be incomplete.

**Review by starbops — APPROVED** [review]
LGTM, just one nit. Thank you.

**Comment by innobead** [comment]
cc @mingshuoqiu

File: docs/install/requirements.md [hunk 2]
```diff
@@ -28,12 +28,14 @@ Harvester nodes have the following hardware requirements and recommendations for
 | Memory           | 32 GB minimum for testing; 64 GB or above required for production                                                                                                                                                               |
 | Disk Capacity    | 250 GB minimum for testing (180 GB minimum when using multiple disks); 500 GB or above required for production                                                                                                                                  |
 | Disk Performance | 5,000+ random IOPS per disk(SSD/NVMe). Management nodes (first 3 nodes) must be [fast enough for etcd](https://www.suse.com/support/kb/doc/?id=000020100) |
-| Network Card     | 1 Gbps Ethernet minimum for testing; 10Gbps Ethernet required for production                                                                                                                       |
+| Network Card     | 1 Gbps Ethernet minimum for testing; 10Gbps Ethernet required for production; suggest to use 2 NICs for the management cluster network, 2 additional NICs for the dedicated VM workload network       |
```
Comment (jillian-maroket) [inline docs/install/requirements.md:31]:
```suggestion
| Network Card     | Minimum Ethernet speed: 1 Gbps for testing, 10 Gbps for production; Recommended network interfaces: 2 NICs for the management cluster network, 2 additional NICs for the dedicated VM workload network |
```

File: docs/install/requirements.md (hunk 1 above)
Comment (jillian-maroket) [inline docs/install/requirements.md:37]:
```suggestion
- Harvester has a [built-in management cluster network](../networking/clusternetwork.md#built-in-cluster-network) (`mgmt`). To achieve high availability and the best performance in production environments, use at least two NICs in each node to set up a bonded NIC for the management network (see step 6 in [ISO Installation](../install/iso-install.md#installation-steps)). You can also create [custom cluster networks](../networking/clusternetwork.md#custom-cluster-network) for VM workloads. Each custom cluster network requires at least two additional NICs to set up a bonded NIC in every involved node of the Harvester cluster. For more information, see [Cluster Network](../networking/clusternetwork.md#concepts).
```

File: docs/install/requirements.md [hunk 3]
```diff
@@ -28,12 +28,14 @@ Harvester nodes have the following hardware requirements and recommendations for
...
+- Harvester has a [built-in management cluster network](../networking/clusternetwork.md#built-in-cluster-network). To achieve the best performance and high-availability. The general practice in production is to use at least 2 NICs in each node to setup a bonded NIC for the management network. Refer the step 6 of the [ISO Installation](../install/iso-install.md#installation-steps). You may also create additional [customized cluster network](../networking/clusternetwork.md#custom-cluster-network) to carry the VM workload network separately. Each of them requires at least 2 additional NICs to setup a bonded NIC in every involved node of the Harvester cluster. You may refer [Cluster Network](../networking/clusternetwork.md#concepts) for more details.
+- For testing purpose, you may use only one NIC for the [built-in management cluster network](../networking/clusternetwork.md#built-in-cluster-network), and even to test the [VM network](../networking/harvester-network.md#create-a-vm-network) which is also carried by this [built-in management cluster network](../networking/clusternetwork.md#built-in-cluster-network). There is no assurance of performance or high-availability.
```
Comment (jillian-maroket) [inline docs/install/requirements.md:38]:
```suggestion
- During testing, you can use only one NIC for the [built-in management cluster network](../networking/clusternetwork.md#built-in-cluster-network) (`mgmt`), and for testing the [VM network](../networking/harvester-network.md#create-a-vm-network) that is also carried by `mgmt`. High availability and optimal performance are not guaranteed.
```

File: docs/networking/clusternetwork.md [hunk 4]
```diff
@@ -85,6 +85,12 @@ If there is no need for traffic separation, you can put all your network under t
 
 You are allowed to add the custom cluster network, which will not be available until it's enabled on some hosts by adding a network configuration.
 
+:::note
+
+Before creating a new cluster network, check the [hardware requirements](../install/requirements.md#hardware-requirements) are met.
```
Comment (jillian-maroket) [inline docs/networking/clusternetwork.md:90]:
```suggestion
Before creating a new cluster network, ensure that the [hardware requirements](../install/requirements.md#hardware-requirements) are met.
```

**Review by jillian-maroket — COMMENTED** [review]
Review done

**Review by bk201 — APPROVED** [review]
lgtm after adapting Jillian's changes.

**Comment by w13915984028** [comment]
@jillian-maroket @Vicente-Cheng 
Besides the comments I also added information about the `witness node` in the second commit, please help take another/a look, thanks.

**Review by starbops — APPROVED** [review]

File: docs/install/requirements.md [hunk 5]
```diff
@@ -28,12 +28,14 @@ Harvester nodes have the following hardware requirements and recommendations for
 | Memory           | 32 GB minimum for testing; 64 GB or above required for production                                                                                                                                                               |
 | Disk Capacity    | 250 GB minimum for testing (180 GB minimum when using multiple disks); 500 GB or above required for production                                                                                                                                  |
 | Disk Performance | 5,000+ random IOPS per disk(SSD/NVMe). Management nodes (first 3 nodes) must be [fast enough for etcd](https://www.suse.com/support/kb/doc/?id=000020100) |
-| Network Card     | 1 Gbps Ethernet minimum for testing; 10Gbps Ethernet required for production                                                                                                                       |
+| Network Card     | Minimum Ethernet speed: 1 Gbps for testing, 10 Gbps for production; Recommended network interfaces: 2 NICs for the management cluster network, 2 additional NICs for the dedicated VM workload network (except the [Witness Node](../advanced/witness.md))       |
```
Comment (jillian-maroket) [inline docs/install/requirements.md:31]:
```suggestion
| Network Card     | Minimum Ethernet speed: 1 Gbps for testing, 10 Gbps for production; Recommended network interfaces: 2 NICs for the management cluster network, 2 additional NICs for the dedicated VM workload network (does not apply to the [witness node](../advanced/witness.md))       |
```

**Review by jillian-maroket — COMMENTED** [review]
Review done

File: docs/install/requirements.md (hunk 5 above)
Comment (jillian-maroket) [reply docs/install/requirements.md:31]: The requested change was not applied.

File: docs/networking/clusternetwork.md [hunk 6]
```diff
@@ -114,4 +128,6 @@ You are allowed to add the custom cluster network, which will not be available u
 
 Starting with Harvester v1.1.2, Harvester supports updating network configs. Make sure to stop all affected VMs before updating network configs.
 
+Recommend to create one `Network Config` for each node or each group of nodes. It will be more convenient when you maintain the cluster like replacing old NICs to new NICs in different slots, you need to stop/migrate the affected VMs and then update the `Network Config`.
```
[the remark this answers is not in the document]
Comment (jillian-maroket) [reply docs/networking/clusternetwork.md:131]: The requested change was not applied.

**Review by jillian-maroket — COMMENTED** [review]
Some requested changes were not applied.

**Review by jillian-maroket — APPROVED** [review]

**Comment by Wizend** [comment]
Hi,

We got kind of a similar problem. In our physical server we got only 1 NIC, but we cannot create a Cluster Network with it because Harvester don't find the NIC. Maybe we just don't understand the explanation in the next issues : https://github.com/harvester/harvester/issues/5509, and the actual one here.

So do we have to add a new network card to the server ? or maybe we have to do some new configurations on our card to get multiple interfaces ?

Excuse our stupid questions, we are students and we try to learn Harvester :p
Thanks for your reply

**Comment by w13915984028** [comment]
> Hi,
> 
> We got kind of a similar problem. In our physical server we got only 1 NIC, but we cannot create a Cluster Network with it because Harvester don't find the NIC. Maybe we just don't understand the explanation in the next issues : [harvester/harvester#5509](https://github.com/harvester/harvester/issues/5509), and the actual one here.
> 
> So do we have to add a new network card to the server ? or maybe we have to do some new configurations on our card to get multiple interfaces ?
> 
> Excuse our stupid questions, we are students and we try to learn Harvester :p Thanks for your reply

At the moment, Harvester doesn't support `channalize` a physical ethernet port.

When your server has only 1 NIC, it is used by the default & mandatory `mgmt` cluster-network.

If you want to create another cluster-network ( sometimes it is called network plane, e.g. management plane, data plane), you need at least one anohter NIC for this network to use. This is for traffic isolation and many others.

If you just want to test `vm network`, you can create it via https://docs.harvesterhci.io/v1.2/networking/harvester-network#create-a-vm-with-vlan-network, which is backed by the mgmt network.

**Review by bk201 — APPROVED** [review]

**Pull request merged**